### PR TITLE
Adds datepicker

### PR DESCRIPTION
**Pull request by chriserik**
screen shot 2016-11-11 at 18 12 49

### CODE REVIEW

File: src/components/Datepicker/Datepicker.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    // Return filled week bins
+    return fullWeekRange.toArray('days').reduce((acc, val) => {
+      if (moment(val).day() % 7 === 0) { acc.push([]); }
+      acc[acc.length - 1].push([moment(val), moment(val).within(monthRange)]);
+      return acc;
+    }, [[]]).filter(bin => bin.length > 0);
+  }
+
+  renderYearOptions() {
+    const prevYears = moment(this.state.viewedDate).subtract(2, 'years');
```
Comment (wwwdata) [inline src/components/Datepicker/Datepicker.js]: the amount of years before and after the current year should be made configurable via a prop

File: src/components/Datepicker/Datepicker.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      const year = moment(momentYear).year();
+      return <option key={year} value={year}>{year}</option>;
+    });
+  }
+
+  renderYearPicker() {
+    return (
+      <div className={this.prefix(['shrink-none'])}>
+        <label className={this.prefix(['assistive-text'])} htmlFor="select-01">Pick a Year</label>
+        <div className={this.prefix(['slds-select_container'])}>
```
Comment (wwwdata) [inline src/components/Datepicker/Datepicker.js]: you call the prefix function but the string already contains `slds` at the front

File: src/components/Datepicker/Datepicker.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  render() {
+    return (
+      <div className={this.prefix(['datepicker', 'dropdown', 'dropdown--left'])}>
+        <div className={this.prefix(['datepicker__filter', 'grid'])}>
+          <div className={this.prefix(['datepicker__filter--month', 'grid', 'grid--align-spread', 'grow'])}>
+            <div className={this.prefix(['align-middle'])}>
+              <Button
+                icon
+                onClick={() => this.viewMonth(-1)}
+                className={this.prefix(['button', 'button--icon-container'])}
```
Comment (wwwdata) [inline src/components/Datepicker/Datepicker.js]: you use our button, which should already have the `button` css class and `icon-container` is also a flavor afaik

File: src/components/Datepicker/Datepicker.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+              </Button>
+            </div>
+            <h2 id="month" className={this.prefix(['align-middle'])}>
+              {this.state.viewedDate.format('MMMM')}
+            </h2>
+            <div className={this.prefix(['align-middle'])}>
+              <Button
+                icon
+                onClick={() => this.viewMonth(1)}
+                className={this.prefix(['button', 'button--icon-container'])}
```
Comment (wwwdata) [inline src/components/Datepicker/Datepicker.js]: another button

File: src/components/Datepicker/Datepicker.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        <table className={this.prefix(['datepicker__month'])} role="grid">
+          {this.renderWeekHeader()}
+          {this.renderMonth()}
+          <tr>
+            <td colSpan="7" role="gridcell">
+              <a
+                onClick={this.resetViewToToday}
+                className={this.prefix(['show--inline-block', 'p-bottom--x-small'])}
+              >
+               Today
```
Comment (wwwdata) [inline src/components/Datepicker/Datepicker.js]: English string. We should also allow to inject the strings via prop and define the English strings as default prop

File: src/components/Datepicker/Datepicker.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    const nextYears = moment(this.state.viewedDate).add(2, 'years');
+    return moment.range(prevYears, nextYears).toArray('years').map((momentYear) => {
+      const year = moment(momentYear).year();
+      return <option key={year} value={year}>{year}</option>;
+    });
+  }
+
+  renderYearPicker() {
+    return (
+      <div className={this.prefix(['shrink-none'])}>
```
Comment (fspoettel) [inline src/components/Datepicker/Datepicker.js]: prefix accepts strings or arrays and handles both correctly, there is no need to pass arrays for single classes

**Comment by chriserik** [comment]
Commits are squashed now.

**Comment by wwwdata** [comment]
> ...by reviewers with write access

looks like my reviews don't count anymore 😂

**Pull request merged**